sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -57,7 +57,9 @@ class GpuProjectExecMeta(
     val gpuExprs = childExprs.map(_.convertToGpu().asInstanceOf[NamedExpression]).toList
     val gpuChild = childPlans.head.convertIfNeeded()
     if (conf.isProjectAstEnabled) {
-      if (childExprs.forall(_.canThisBeAst)) {
+      // cuDF requires return column is fixed width
+      val allReturnTypesFixedWidth = gpuExprs.forall(e => GpuBatchUtils.isFixedWidth(e.dataType))
+      if (childExprs.forall(_.canThisBeAst) && allReturnTypesFixedWidth) {
```

Comment:
Micro optimization: use the already-computed boolean value first, so the `if` condition can short-circuit if it is false.
```suggestion
      if (allReturnTypesFixedWidth && childExprs.forall(_.canThisBeAst)) {
```